students/CCSt130/lesson03/mail_room_pt1.py

```diff
@@ -0,0 +1,500 @@
+""" This code allows a user to update a list of donors, print
+all donations, average donations, and print a thank you email. 
+"""
+
+"""
+Lesson03 :: Mailroom Part 1
+@author: Chuck Stevens :: CCSt130
+Created on Sun Jun 16 19:32:15 2019
+"""
+
+import datetime # dates needed for comparison
+
+from collections import Counter
+
+from statistics import mean
+
+if __name__ == "__main__":
+    
+    def main():
+        """ Contains list of donors and displays Menu. """
+        all_donations = [[2018,8,26,"Bill and Melinda Gates Foundation",48405000], \
+                      [2014,8,14,"Stichting INGKA Foundation",40338000], \
+                      [2013,7,12,"Stichting INGKA Foundation",31837000], \
+                      [2014,8,16,"Buffet Foundation",27940000], \
+                      [2014,8,17,"Bill and Melinda Gates Foundation",26340000], \
+                      [2017,8,19,"Bill and Melinda Gates Foundation",23049002], \
+                      [2016,8,19,"Bill and Melinda Gates Foundation",22737329], \
+                      [2018,8,26,"Bill and Melinda Gates Foundation",21455000], \
+                      [2016,8,20,"Bill and Melinda Gates Foundation",20670300], \
+                      [2019,2,11,"Bezos Family Foundation",18938000], \
+                      [2015,2,6,"Bezos Family Foundation",19554000], \
+                      [2014,6,27,"Stichting INGKA Foundation",19474000], \
+                      [2016,8,20,"Walton Family Foundation",18947775], \
+                      [2015,8,13,"Buffet Foundation",18603000], \
+                      [2016,3,11,"Walton Family Foundation",17914000], \
+                      [2015,8,17,"Walton Family Foundation",17789000], \
+                      [2015,8,17,"Walton Family Foundation",17440000], \
+                      [2011,8,21,"Walton Family Foundation",18254000], \
+                      [2017,8,18,"Stichting INGKA Foundation",15966000], \
+                      [2014,8,16,"Walton Family Foundation",16066000], \
+                      [2017,8,19,"Walton Family Foundation",14841309], \
+                      [2013,11,21,"Buffet Foundation",15381000], \
+                      [2015,12,10,"Buffet Foundation",15115000], \
+                      [2017,8,19,"Walton Family Foundation",14391572], \
+                      [2015,8,13,"Buffet Foundation",14534000], \
+                      [2016,8,19,"Bill and Melinda Gates Foundation",14354375], \
+                      [2016,8,20,"Walton Family Foundation",14093386]]
+        
+        usr_input = ""
+        
+        while True:
+            # Displaying menu header
+            print()
+            print(48 * '*')
+            print()
+            print(17 * '*', "Menu Options", 17 * '*')
+            print()
+
+            # Display menu options
+            print(" 1. Create a Report of all Donations")
+            # print(" 2. Print Thank You Emails for all Donors") TBA
+            print(" 2. Print a List of Donors' Donation Averages")
```

Comment:
option # 2 should be printing report (shown example output below) sorted by total given